build.sbt

```diff
@@ -53,27 +42,53 @@ lazy val commonSettings = Seq(
   publishTo := sonatypePublishTo.value
 )
 
+def commonScalacOptions(scalaVersion: String) = {
```

Comment:
what do you think about https://tpolecat.github.io/2017/04/25/scalac-flags.html ?

Reply:
I thought about adding those, but perhaps that could be a separate PR as it is much more restrict than what we have now.

Reply:
agree :)